spinalcordtoolbox/scripts/sct_smooth_spinalcord.py

```diff
@@ -210,11 +209,19 @@ def main(args=None):
 
     # Smooth the straightened image along z
     printv('\nSmooth the straightened image...')
-    sigma_smooth = ",".join([str(i) for i in sigma])
-    sct_maths.main(args=['-i', 'anat_rpi_straight.nii',
-                         '-smooth', sigma_smooth,
-                         '-o', 'anat_rpi_straight_smooth.nii',
-                         '-v', '0'])
+
+    img = Image("anat_rpi_straight.nii")
+    out = img.copy()
+
+    if len(sigmas) == 1:
+        sigmas = [sigmas[0] for i in range(len(img.data.shape))]
+    elif len(sigmas) != len(img.data.shape):
+            raise ValueError("-smooth need the same number of inputs as the number of image dimension OR only one input")
+
+    sigmas = [sigmas[i] / img.dim[i + 4] for i in range(3)]
```

Comment:
Same comment -- maybe this sigma adjustment should be placed within `math.smooth` instead.